### PR TITLE
Improve the signing process and installer display

### LINKED ISSUE #84: Improve the signing process and installer display
For context see this [Slack convo](https://learningequality.slack.com/messages/CB37UM23A/convo/CB37UM23A-1533577830.000452/).

1. Display the **_signed_** Windows installer on the [tagged release GH page](https://github.com/learningequality/kolibri/releases) together with the rest of the builds (PEX, ZIP...)? We now have to hunt for it on Buildkite and it can induce the confusion...

2. Configure Buildkite to produce the installer with the filename that clearly indicates that the installer is signed. It could help to distinguish the intermediate builds from PRs that are not signed.

cc @indirectlylit @aronasorman

### PR DESCRIPTION
**Pull request by mrpau-richard**
### Summary
I automatically rename the signed Windows Installer after the build finish at the Buildkite. The reason of this, is to easily know if the installer is sign during the testing.

### Reviewer guidance
* Download the signed Windows installer [here](https://buildkite.com/learningequality/mrpau-windows-sign-testing/builds/61#567b6eca-df4e-479e-888a-d8e6f9c4e320).
* Try to install and run it at the Windows VM and see if the Kolibri server is properly running, and its version is correct.

### References
This fixes [#84](https://github.com/learningequality/kolibri-installer-windows/issues/84)

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [x] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by indirectlylit** [comment]
looks like there's a merge conflict. likely related to @benjaoming's recent docker work in #4056

**Comment by indirectlylit** [comment]
@benjaoming would you mind reviewing this PR?

**Comment by benjaoming** [comment]
Hi @mrpau-richard and @radinamatic 

LMK if you disagree, but I think we need to add a "-unsigned" suffix to ".exe" files that are unsigned. Because when a list of files is displayed and there is no indication of whether a file is signed or not, then it's hard to know that signed files exist, unless you have seen them somewhere else.

So I'm suggesting: 1) Name the existing .exe file "kolibri-blahblah-UNSIGNED.exe" and the signed one as just "kolibri-blahblah.exe" (you can of course add "-SIGNED", but I think the actual nomenclature has to happen because a file *isn't* signed).

**Comment by mrpau-richard** [comment]
>So I'm suggesting: 1) Name the existing .exe file "kolibri-blahblah-UNSIGNED.exe" and the signed one as just "kolibri-blahblah.exe" (you can of course add "-SIGNED", but I think the actual nomenclature has to happen because a file isn't signed).

@benjaoming I agree with it. Instead of renaming the installer that's signed, lets rename only the unsigned installer.

**Comment by mrpau-richard** [comment]
@radinamatic I have this sample output of the unsigned Windows installer filename [here](https://buildkite.com/learningequality/mrpau-windows-sign-testing/builds/71#be0d117e-ea3c-4073-99a1-6d85e0a310d9), And the signed one is [here](https://buildkite.com/learningequality/mrpau-windows-sign-testing/builds/71#9975e8f0-256c-4f43-9e41-638eccaeb415). please check. Thanks.

**Comment by benjaoming** [comment]
Hi @mrpau-richard - could you trigger the signing step in a commit in this PR?

**Comment by indirectlylit** [comment]
Hey, this is looking good. Two comments:

1. I believe the windows installer signing should be behind a buildkite block step, correct? It looks like with the current process it will sign every installer?
2. It looks like currently the signed artifacts don't get uploaded?

File: .buildkite/sign_windows_installer.bat [hunk 1]
```diff
@@ -1,13 +1,13 @@
 @echo off
-rem This will download the Kolibri windows installer artifact at the windows-2016 buildkite agent.
-rem After the installer successfully sign it will upload the signed installer at the Sign Windows installer pipeline artifact.
-rem To sign the Windows installer set the `SIGN_WINDOWS_INSTALLER=true` environment variable.
+rem To sign the Windows installer, set the `SIGN_WINDOWS_INSTALLER=true` environment variable.
+rem This will download the Kolibri Windows installer artifact at the windows-2016 Buildkite agent.
```
Comment (cpauya) [inline .buildkite/sign_windows_installer.bat]: @mrpau-richard Can you re-phrase these last two lines to make it simpler?

**Review by cpauya — DISMISSED** [review]

File: .buildkite/sign_windows_installer.bat [hunk 2]
```diff
@@ -1,13 +1,13 @@
 @echo off
-rem This will download the Kolibri windows installer artifact at the windows-2016 buildkite agent.
-rem After the installer successfully sign it will upload the signed installer at the Sign Windows installer pipeline artifact.
-rem To sign the Windows installer set the `SIGN_WINDOWS_INSTALLER=true` environment variable.
+rem To sign the Windows installer, set the `SIGN_WINDOWS_INSTALLER=true` environment variable.
```
Comment (cpauya) [inline .buildkite/sign_windows_installer.bat]: If this is an OS env var, I think it's good to start it with a prefix like `KOLIBRI_SIGN_WINDOWS_INSTALLER`... just like `BUILDKITE_BUILD_ID`, `BUILDKITE_AGENT_ACCESS_TOKEN`... not important but nice to have.

**Comment by benjaoming** [comment]
@mrpau-richard - if you look at the `pipeline.yml`, you'll see what is meant by "Block step". Documentation here: https://buildkite.com/docs/pipelines/block-step

**Comment by mrpau-richard** [comment]
@benjaoming Thanks to your help. It's now using the Builkite block step and automatically upload the Kolibri signed Windows installer artifact. 

I will create another PR to update the Windows signing doc [here](https://github.com/learningequality/kolibri-installer-windows/tree/develop/windows/kolibri-signing-doc).

**Comment by benjaoming** [comment]
@mrpau-richard 

Can you make it so that the signed .exe is called `-signed` at the end? I.e. this should be `kolibri-0.11.0a7.dev0+git.88.ge1750c76-signed.exe`


image

Next, it needs to be uploaded to Github -- try changing how [upload_artifacts.py](https://github.com/learningequality/kolibri/blob/develop/.buildkite/upload_artifacts.py) works. Currently, it will upload all detected files of known extension names.. but just before the block steps start. That's because we don't want uploading to be a block step, that way, we would have to unblock it every time.

But could you change `upload_artifacts.py` so it can be called many times and only upload files once? For instance, you could append the file name to `upload.manifest` and use buildkite artifacts to save the file after uploading.. then when calling `upload_artifacts.py` again a second time, it could read and ignore files already listed.

**Comment by mrpau-richard** [comment]
@benjaoming I'll defer working on this PR for now, and prioritise the kolibri-0.11.0-beta integration testing.

**Comment by benjaoming** [comment]
@rtibbles is this still destined for 0.11.x? We could re-target the PR to `develop` again and the long list of `develop` commits _should_ vanish...

**Comment by indirectlylit** [comment]
nice - it's getting there!

**Comment by mrpau-richard** [comment]
@benjaoming I changed the target branch to develop. 

I have these sample output how the unsigned and signed Windows installer display at the report  `html` 
https://storage.googleapis.com/le-downloads/kolibri-release-9625-first-report.html
https://storage.googleapis.com/le-downloads/kolibri-release-9625-second-report.html

**Review by benjaoming — APPROVED** [review]
All of this looks to be working to me, and I was able to generate a signed windows installer in Buildkite just now. I cannot see any further work to be done here, and @radinamatic was also happy with this :+1: 

Great work @mrpau-richard - glad to have this in!

**Comment by radinamatic** [comment]
Yey!!! 👍 💥 🎉

**Comment by indirectlylit** [comment]
thanks all!

**Pull request merged**